AEPAssurance/Source/AssuranceEventChunker.swift

```diff
@@ -13,28 +13,57 @@
 struct AssuranceEventChunker {
-    let CHUNK_SIZE = 30 * 1024 // 30KB
 
+    /// The maximum size of data that an `AssuranceEvent` payload can hold after chunking
+    ///
+    /// How did we derive to 30KB?
+    ///  The maximum size of an `AssuranceEvent` to get successfully delivered through the socket is 32KB.
+    ///  AssuranceEvent consist of payload (Dictionary), type(String), vendor(String), metadata (Dictionary), timestamp(Long) and EventNumber (Integer)
+    ///  For the AssuranceEvent to completely fit into the maximum allowed socket size, we safely assign
+    ///    30KB for payload
+    ///    2KB for other fields
+    ///
+    /// Accounting for escape string bloat factor:
+    ///   After chunking the payload into multiple consumable data size, the chunk's are then recreated into corresponding `AssuranceEvent`'s.
+    /// During this process the chunked payload string is escaped and put inside the `chunkData` field of the resulting `Assurance Event`. This
+    /// escaping of string further increases the size of the chunked Data. To accommodate for unknown bloating factor, the chunk size for each event is reduced to 15KB.
+    /// Additionally the factor 0.75 corresponds to size correction due to base64 Encoding of data before sending them over Websocket.
+    let CHUNK_SIZE = (Int) ((15 * 1024) * 0.75) // 15KB
+
+    /// Chunks the given `AssuranceEvent` into multiple socket consumable size AssuranceEvents
+    ///
+    /// The payload field in the `AssuranceEvent` structure has the potential to bottleneck the size limit. Hence only the payload is chopped into multiple smaller chunks.
+    /// Once the payload is chunked, then the chunked data is added in the payload of each AssuranceEvent under the key "chunkData".
+    /// And chunked details are added to the metadata field of the Assurance Event. The chunk details comprises of
+    ///   1. chunkId - Unique Id representing all the chunks of a single event.
+    ///   2. chunkTotal - The total number of chunks to define the original event
+    ///   3. chunkSequenceNumber - Integer Value representing the sequence of chunks
+    ///
+    /// - Parameter event: An `AssuranceEvent` that is over the socket size limit that needs to be chunked
+    /// - Returns: An array of chunked AssuranceEvents
     func chunk(_ event: AssuranceEvent) -> [AssuranceEvent] {
         var chunkedEvents: [AssuranceEvent] = []
 
+        /// Highly unlikely to have an event that needs to be chunked without a payload. If so this is an misbehaving event
+        /// We should discard this event and return an empty array in such cases
         guard let eventPayload = event.payload else {
-            return chunkedEvents
+            Log.warning(label: AssuranceConstants.LOG_TAG, "Discarding the Assurance Event that is demanding to be chunked without a payload. \(event.description)")
+            return []
```

Comment:
maybe it's best to return `[event]` here instead?

Reply:
Response to this and the previous comment

The decision if an event needs to be chunk happens in the [Websocket](https://github.com/adobe/aepsdk-assurance-ios/blob/dev/AEPAssurance/Source/Socket/WebView/WebViewSocket.swift#L122) code. 
We designed Chunker to be only responsible for chunking large events. Since the decision is made before calling the chunk function, any event that is passed inside the chunker is automatically assumed to be a large event.


Answer to the current comment (Reason for returning empty array)

The expectation of a chunker is to return a set of events that are socket-consumable. And since only large-sized events are passed to the chunker and in this case, if the payload for such event is nil, this seems like a misbehaving event. In such cases, it's better to log and drop this event rather than having it passed to the WebSocket (which disconnects the connection with 1009 error for TOO BIG DATA)

https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent/code